src/main/java/eu/dissco/backend/repository/MasJobRecordRepository.java

```diff
@@ -77,12 +94,24 @@ public List<MasJobRecordFull> getMasJobRecordsByCreatorAndState(String creatorId
         .fetch(this::recordToMasJobRecord);
   }
 
+  public List<MasJobRecordFull> getMasJobRecordsByUserIdAndState(String userId, String state,
+      int pageNum, int pageSize) {
+    var offset = getOffset(pageNum, pageSize);
+    return context.select(MAS_JOB_RECORD.asterisk())
+        .from(MAS_JOB_RECORD)
+        .join(NEW_USER)
+        .on(NEW_USER.ORCID.eq(MAS_JOB_RECORD.USER_ID))
+        .where(NEW_USER.ID.eq(userId))
+        .and(MAS_JOB_RECORD.STATE.eq(state))
```

Comment:
Made some changes for a dynamic condition. 

Jooq documentation has every standalone condition starting with `noCondition()` so it looks a bit funny but works better than what we had imo.  
https://www.jooq.org/doc/latest/manual/sql-building/dynamic-sql/

Reply:
It does look better but I don't think the `noCondition()` is necessary. Why not just do `where()`? It looks like the first condition is always there so that can be the where and then possibly extended by the `and`. To me it looks like the noCondition is for when there might be a condition but you are not certain, here we are certain.

Reply:
Fixed :tada:  looks like we don't even need `where()`

Reply:
With the SQL condition in the PR.
What I think does happen now is that you get a nested condition, so `WHERE (userid = X AND state = y)` instead of a `WHERE userid = X AND state = y`
Don't think this is an issue in this case, needed this nesting somewhere but I forget where